exercises/practice/react/lib/react.ex

```diff
@@ -0,0 +1,41 @@
+defmodule React do
+  @doc """
+  Start a reactive system
+  """
+  @spec new(cells :: [%{atom => any}]) :: {:ok, pid}
+  def new(cells) do
+  end
+
+  @doc """
+  Return the value of an input or output cell
+  """
+  @spec get_value(cells :: pid, cell :: String.t()) :: any()
+  def get_value(cells, cell) do
+  end
+
+  @doc """
+  Set the value of an input cell
+  """
+  @spec set_value(cells :: pid, cell :: String.t(), value :: any) :: :ok
+  def set_value(cells, cell, value) do
+  end
+
+  @doc """
+  Add a callback to an output cell
+  """
+  @spec add_callback(
+          cells :: pid,
+          cell :: String.t(),
+          callback :: String.t(),
+          send :: (String.t(), any -> :ok)
```

Comment:
Does it really matter what the callback returns?

Reply:
Well the callback must send a message, which is of type `:ok`, otherwise the tests can't be passed.

Reply:
Must send a message? The callback we use in tests sends a message so that we have an easy way of verifying that it was called, but in general the callback can do anything. It's even in your implementation:

```
Enum.each(cell.callbacks, fn {name, send} -> send.(name, value) end)
```

`Enum.each` -> return value gets discarded, only side effects matter

Reply:
Right, I get it. 

I think what I was doing here is matching the type of the actually callbacks , which are all predefined in the tests as `send_callback` and are indeed all `send`, but I get what you are saying, in a real API, you wouldn't restrict the callback type. I will change it :)